src/models/donation.rb

```diff
@@ -1,8 +1,11 @@
 class Donation < ActiveRecord::Base
   validates_presence_of :shop, :order_id, :donation_amount
   validates_uniqueness_of :order_id, scope: :shop, conditions: -> { where("status != 'void' or status is null") }
+  validates_uniqueness_of :donation_number, scope: :shop
```

Comment:
this validation may fail for voiding reciepts because the update will not pass validation. These jobs (if there are any) will be retried after the backfill